modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/ClayCriteriaBuilder.es.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClayCriteriaGroup from './ClayCriteriaGroup.es';
+import ClayButton from './ClayButton.es';
+
+class ClayCriteriaBuilder extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			editing: false,
+			initialCriteria: this.props.criteria
+		};
+	}
+
+	render() {
+		const {conjunctions, criteria, operators, properties} = this.props;
+
+		const {editing} = this.state;
+
+		return (
+			<div className="criteria-builder">
+				<div className="criteria-builder-toolbar">
+					<ClayButton
+						label={Liferay.Language.get('edit')}
+						onClick={this._handleToggleEdit}
+					/>
+				</div>
+
+				{this._isCriteriaEmpty() ? (
+					<ClayCriteriaGroup
+						conjunctions={conjunctions}
+						criteria={criteria}
+						criteriaTypes={ClayCriteriaBuilder._buildCriteriaTypes(
+							operators
+						)}
+						editing={editing}
+						onChange={this._updateCriteria}
+						operators={operators}
+						properties={properties}
+						root
+					/>
+				) : (
+					<div
+						className="empty-state"
+						onClick={this._handleNewCriteria}
+					>
+						{Liferay.Language.get('click-to-start-editing')}
+					</div>
+				)}
+			</div>
+		);
+	}
+
+	static _buildCriteriaTypes(operators) {
+		return operators.reduce(
+			(criteriaTypes, {supportedTypes}) => {
+				supportedTypes.forEach(
+					type => {
+						if (!criteriaTypes[type]) {
+							criteriaTypes[type] = operators.filter(
+								operator =>
+									operator.supportedTypes.includes(type)
+							);
+						}
+					}
+				);
+
+				return criteriaTypes;
+			},
+			new Map()
+		);
+	}
+
+	_cleanCriteria(criterion) {
+		const test = criterion
+			.filter(
+				({items}) => {
+					return items ? items.length : true;
+				}
+			)
+			.map(
+				item => {
+					return item.items ?
+						Object.assign(
+							item,
+							{
+								items: this._cleanCriteria(item.items)
+							}
+						) :
+						item;
+				}
+			);
+
+		return test;
+	}
+
+	_handleNewCriteria = () => {
+		const {onChange, operators, properties} = this.props;
+
+		const emptyItem = {
+			operatorName: operators[0].name,
+			propertyName: properties[0].name,
+			value: ''
+		};
+
+		onChange(
+			{
+				conjunctionName: 'and',
+				items: [emptyItem]
+			}
+		);
+	}
+
+	_handleToggleEdit = event => {
+		event.preventDefault();
+
+		this.setState(
+			{
+				editing: !this.state.editing
+			}
+		);
+	};
+
+	_isCriteriaEmpty = () => {
+		const {criteria} = this.props;
+
+		return criteria ? criteria.items.length : false;
+	}
+
+	_updateCriteria = newCriteria => {
+		this.props.onChange(this._cleanCriteria([newCriteria]).pop());
```

Comment:
In this case it doesn't since the only value needed is the first item. Maybe if I time (in another ticket) it could be neat to use immutable or these immutability helpers (https://reactjs.org/docs/update.html).